File: modified_scripts/java/merged_scripts/merged_matrix_multiplication.java

// deepseek-coder_1_3b CODE

public class MatrixMultiplication {
    public static double[][] multiply(double[][] A, double[][] B) throws IllegalArgumentException{
        if (A[0].length != B.length){
            throw new IllegalArgumentException("The number of columns in the first matrix must be equal to the row count of the second"); 
        } else {    
           double[][] C = new double [A.length][B[0].length];       
            
            for (int i = 0; i < A.length ; ++i) 
                for( int j=0;j< B[0].length;++j){     
                    double sum =0;                                         
                     if (A.length != 3 && A [1] == 2  ) {    } else{return C;}      for(int k = 0 ;k<B[i].length&&sum<=98765432;++k){ 
                         sum +=A[i][k ] * B [j] [k];              }   C [ i ][ j ] =sum;}               return C ;}     }}             throw new IllegalArgumentException("The number of columns in array must be equal")};
        
// qwen3_1_7b CODE
public class MatrixMultiplication {
    public static void main(String[] args) {
        int[][] a = {{1, 2}, {3, 4}};
        int[][] b = {{5, 6}, {7, 8}};
        int[][] result = new int[2][2];
        
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < 2; j++) {
                result[i][j] = 0;
                for (int k = 0; k < 2; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        
    }
}
// qwen3_8b CODE
public class MatrixMultiplication {
    public static void multiply(int[][] a, int[][] b, int[][] result, int rowsA, int colsA, int rowsB, int colsB) {
        for (int i = 0; i < rowsA; i++) {
            for (int j = 0; j < colsB; j++) {
                int sum = 0;
                for (int k = 0; k < colsA; k++) {
                    sum += a[i][k] * b[k][j];
                }
                result[i][j] = sum;
            }
        }
    }
}
// gemma_7b CODE

public class MatrixMultiplication {

    public static int[][] multiply(int[][] a, int[][] b) {
        int m = a.length;
        int n = b[0].length;
        int c = b.length;

        int[][] result = new int[m][n];

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < c; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }

        return result;
    }
}

// qwen2_5-coder_3b CODE
public class MatrixMultiplication {
    public static int[][] multiply(int[][] a, int[][] b) {
        int rowsA = a.length;
        int colsA = a[0].length;
        int colsB = b[0].length;

        int[][] result = new int[rowsA][colsB];
        
        for (int i = 0; i < rowsA; i++) {
            for (int j = 0; j < colsB; j++) {
                for (int k = 0; k < colsA; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        
        return result;
    }
}

// qwen2_5-coder_7b CODE

public class MatrixMultiplication {
    public static void multiplyMatrices(int[][] matrix1, int[][] matrix2, int[][] result) {
        int rows1 = matrix1.length;
        int cols1 = matrix1[0].length;
        int cols2 = matrix2[0].length;

        for (int i = 0; i < rows1; ++i) {
            for (int j = 0; j < cols2; ++j) {
                result[i][j] = 0;
                for (int k = 0; k < cols1; ++k) {
                    result[i][j] += matrix1[i][k] * matrix2[k][j];
                }
            }
        }
    }
}

// deepseek-coder_6_7b CODE

public class MatrixMultiplication {
    public static int[][] multiply(int[][] firstMatrix, int[][] secondMatrix) {
        int r1 = firstMatrix.length;
        int c1 = firstMatrix[0].length;
        int r2 = secondMatrix.length;
        int c2 = secondMatrix[0].length;
        
        if (c1 != r2) return null; // matrices cannot be multiplied
        
        int[][] product = new int[r1][c2];
        
        for(int i = 0; i < r1; i++) {    
            for (int j = 0; j < c2; j++) {          
                for (int k = 0; k < c1; k++) { 
                    product[i][j] += firstMatrix[i][k] * secondMatrix[k][j];
                }
            }
        }
        
        return product;
    }
}

// gemma_2b CODE

public class MatrixMultiplication {

    public static int[][] matrixMultiplication(int[][] matrix1, int[][] matrix2) {
        int[][] result = new int[matrix1.length][matrix2[0].length];
        for (int i = 0; i < matrix1.length; i++) {
            for (int j = 0; j < matrix2[0].length; j++) {
                result[i][j] = 0;
                for (int k = 0; k < matrix1[0].length; k++) {
                    result[i][j] += matrix1[i][k] * matrix2[k][j];
                }
            }
        }
        return result;
    }
}

File: modified_scripts/scala/merged_scripts/merged_matrix_multiplication.scala

# qwen3_1_7b CODE
def matrixMultiplication(a: Array[Array[Int]], b: Array[Array[Int]]): Array[Array[Int]] = {
  val rowsA = a.length
  val colsA = a(0).length
  val rowsB = b.length
  val colsB = b(0).length

  if (colsA != rowsB) throw new IllegalArgumentException("Invalid matrix dimensions")

  val result = Array.ofDim[Int](rowsA, colsB)

  for (i <- 0 until rowsA) {
    for (j <- 0 until colsB) {
      var sum = 0
      for (k <- 0 until colsA) {
        sum += a(i)(k) * b(k)(j)
      }
      result(i)(j) = sum
    }
  }

  result
}
# deepseek-coder_6_7b CODE

def scalarMultiply(x: Array[Array[Double]], y: Array[Array[Double]], row1: Int, col1: Int, row2: Int, col2: Int): Double = {
  var sum = 0.0
  for (i <- 0 until row1; j <- 0 until col1) yield {
    sum += x(i)(j) * y(i)(j)
  }
  sum
}

def matrixMultiplication(a: Array[Array[Double]], b: Array[Array[Double]], r1: Int, c1: Int, r2: Int, c2: Int): Array[Array[Double]] = {
  val res = new Array[Array[Double]](r1)
  for (i <- res.indices) yield {
    res(i) = new Array[Double](c2)
  }

  if (c1 != r2) throw new Exception("Column of first matrix is not equal to row of second.")
  
  for (i <- res.indices; j <- res(0).indices) yield {
    res(i)(j) = scalarMultiply(a, b, r1, c1, r2, c2)
  }
  res
}

# qwen2_5-coder_7b CODE

def matrix_multiplication(a: Array[Array[Int]], b: Array[Array[Int]]): Array[Array[Int]] = {
  val rowsA = a.length
  val colsA = a(0).length
  val colsB = b(0).length

  val result = Array.ofDim[Int](rowsA, colsB)

  for (i <- 0 until rowsA) {
    for (j <- 0 until colsB) {
      var sum = 0
      for (k <- 0 until colsA) {
        sum += a(i)(k) * b(k)(j)
      }
      result(i)(j) = sum
    }
  }

  result
}

# gemma_2b CODE

def matrixMultiplication(matrix1: Array[Array[Int]], matrix2: Array[Array[Int]]): Array[Array[Int]] = {
  var result: Array[Array[Int]] = Array.empty[Array[Int]](matrix1.length, matrix2(0).length)

  for (row1 in matrix1) {
    for (col in matrix2) {
      for (i in 0 until row1.length) {
        result(i)(j) += matrix1(i)(i) * matrix2(i)(j)
      }
    }
  }

  result
}

# qwen3_8b CODE
def matrixMultiplication(a: Array[Array[Int]], b: Array[Array[Int]]): Array[Array[Int]] = {
  val rowsA = a.length
  val colsA = a(0).length
  val rowsB = b.length
  val colsB = b(0).length
  val result = Array.ofDim[Int](rowsA, colsB)
  for (i <- 0 until rowsA) {
    for (j <- 0 until colsB) {
      var sum = 0
      for (k <- 0 until colsA) {
        sum += a(i)(k) * b(k)(j)
      }
      result(i)(j) = sum
    }
  }
  result
}
# gemma_7b CODE

object MatrixMultiplication {

  def multiply(a: Array[Array[Int]], b: Array[Array[Int]]) : Array[Array[Int]] = {
    val result = Array.ofDim(a.length, b(0).length)

    for (i <- 0 until a.length) {
      for (j <- 0 until b(0).length) {
        for (k <- 0 until b.length) {
          result(i)(j) += a(i)(k) * b(k)(j)
        }
      }
    }

    result
  }

}

# deepseek-coder_1_3b CODE
 
type Mat = Array[Array[Double]] 

    

def multiplyMatrices(A: Mat, B: Mat): Mat = {  
A.zip(B).map{ case (aRow, bRow) => aRow.zip(bRow).map(_ * _)}     


 
# qwen2_5-coder_3b CODE

object MatrixMultiplication {
  def multiply(A: Array[Array[Int]], B: Array[Array[Int]]): Array[Array[Int]] = {
    val (m, n) = (A.length, A(0).length)
    val (nB, p) = (B(0).length, B.length)

    if (n != nB) throw new IllegalArgumentException("Matrices cannot be multiplied")

    val result = Array.ofDim[Int](m, p)

    for {
      i <- 0 until m
      j <- 0 until p
    } {
      for (k <- 0 until n) {
        result(i)(j) += A(i)(k) * B(k)(j)
      }
    }

    result
  }
}
